### PR TITLE
Move task subclasses into their own files

### LINKED ISSUE #11007: Fix single-table inheritance complaints.
Move `MailTask` and `HearingAdminActionTask` subclasses into their own files. Related [slack discussion](https://dsva.slack.com/archives/C2ZAMLK88/p1558735071019000).

### PR DESCRIPTION
**Pull request by lowellrex**
Resolves #11007.

### CODE REVIEW

File: app/models/tasks/clear_and_unmistakeable_error_mail_task.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class ClearAndUnmistakeableErrorMailTask < MailTask
```
Comment (aroltsch) [inline app/models/tasks/clear_and_unmistakeable_error_mail_task.rb]: shall we create a mail_tasks directory under tasks and put all these guys there?

Comment (lowellrex) [reply app/models/tasks/clear_and_unmistakeable_error_mail_task.rb]: I have a slight preference towards all the tasks being in the same directory (`/tasks`) just for simplicity's sake, but if you feel strongly that we should put them in their own directory then we can do that too.

If we put these classes in their own directory, would we add a `MailTasks::` prefix to their names or would they stay as-is?

Comment (pkarman) [reply app/models/tasks/clear_and_unmistakeable_error_mail_task.rb]: if you change the class name, they would need a new dir.

if you keep the class names the same, they could just go in `/tasks`

I prefer the latter personally.

File: app/models/tasks/hearing_related_mail_task.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class HearingRelatedMailTask < MailTask
```
Comment (aroltsch) [inline app/models/tasks/hearing_related_mail_task.rb]: Can we move hearing tasks under `hearing_tasks` directory the same as `mail_tasks`?

File: app/models/tasks/clear_and_unmistakeable_error_mail_task.rb (hunk 1 above)
Comment (aroltsch) [reply app/models/tasks/clear_and_unmistakeable_error_mail_task.rb]: I don't have a strong preference.

**Pull request merged**